src/utils/jsonValidator.js

```diff
@@ -0,0 +1,78 @@
+const Joi = require('joi')
+
+const schema = Joi.object({
+  name: Joi.string().min(3).max(30).required(),
+  title: Joi.string().min(3).max(30).required(),
+  layout: Joi.string().min(3).max(30).required(),
+  caption: Joi.string().min(3).required(),
+  enabled: Joi.boolean().required(),
+  subCaption: Joi.string().min(3),
+  questions: Joi.array()
+    .required()
+    .items(
+      Joi.object({
```

Comment:
It would be cool if this declaration would be in a separate file (maybe QuestionSchemas could fit) and reused on dependentQuestions instead of duplicating 😬 

Also, I have read the README and it looks like there are different kind of questions with different properties. It would be amazing if each kind of question will have it own schema. 

Something like this:

```javascript
// baseQuestionSchema.js

const question = Joi.object({
  // This way it can have recursive validations
  dependentQuestion: Joi.array().items(Joi.lazy(() => question))
})

// checkBoxSchema.js
import questionSchema from './baseQuestionSchema.js'

const checkBoxSchema = questionSchema.append({
// checkBox specific properties

})

// questionsSchema.js
const questionsSchema = Joi.array().items(CheckBoxSchema, CountrySchema, InputSchema, ...etc)

// jsonValidator.js

questions: questionsSchema.required()

```

This way, each Question update would be independent of the others and they would be easier to test 😬 

What do you believe?

Reply:
I'm loving this approach, so well structured!

Reply:
I've changed the approach since `Joi.lazy()` and `Joi.link()` were not a solution, however I still have a bug I'm currently trying to solve